src/main/java/org/opensearch/sdk/BaseExtension.java

```diff
@@ -35,10 +36,40 @@ public abstract class BaseExtension implements Extension {
     protected ThreadPool threadPool;
 
     /**
-     * Empty constructor to fulfill abstract class requirements
+     * Optional classpath-relative path to a yml file containing extension settings.
+     */
+    protected static final String EXTENSION_SETTINGS_PATH = "/sample/helloworld-settings.yml";
+
+    /**
+     * The extension settings include a name, host address, and port.
+     */
+    protected ExtensionSettings settings;
+
+    /**
+     * Instantiate this extension, initializing the connection settings and REST actions.
      */
     protected BaseExtension() {
+        try {
+            this.settings = initializeSettings();
+        } catch (IOException e) {
+            throw new ExceptionInInitializerError(e);
+        }
+    }
 
+    /**
+     * The Extension must provide its settings to the ExtensionsRunner.
+     * These may be optionally read from a YAML file on the class path.
+     * Or you may directly instantiate with the ExtensionSettings constructor.
+     *
+     * @return This extension's settings.
+     * @throws IOException on failure to load settings.
+     */
+    protected static ExtensionSettings initializeSettings() throws IOException {
```

Comment:
You can't inherit `static` methods so this should definitely not be `protected`. But I think this code is going to be moved into the constructor anyway (along with the javadoc outlining the two options).